packages/app/src/cli/services/context.ts

```diff
@@ -183,14 +185,48 @@ export async function ensureDevContext(options: DevContextOptions, token: string
 
   if (selectedApp.apiKey === cachedInfo?.appId && selectedStore.shopDomain === cachedInfo.storeFqdn) {
     const packageManager = await getPackageManager(options.directory)
-    showReusedValues(organization.businessName, cachedInfo, packageManager)
+    showReusedValues({org: organization.businessName, cachedAppInfo: cachedInfo, packageManager})
   }
 
   const result = buildOutput(selectedApp, selectedStore, useCloudflareTunnels, cachedInfo)
   await logMetadataForLoadedDevContext(result)
   return result
 }
 
+export async function ensureDevContextWithConfig(options: DevContextOptions, token: string): Promise<DevContextOutput> {
+  const localApp = await load({directory: options.directory, specifications: [], configName: options.config})
+
+  if (!localApp.configuration.client_id) {
+    throw new AbortError(`Configuration file ${options.config} needs a client_id.`)
+  }
+
+  const remoteApp = await appFromId(localApp.configuration.client_id, token)
+
+  if (!remoteApp) {
+    throw new AbortError(`Could not find app for client_id: ${localApp.configuration.client_id}`)
+  }
+
+  const orgId = getOrganization() || remoteApp.organizationId
```

Comment:
If we have a `remoteApp`, we should use the org from that app, `getOrganization` might be a different org